### PR TITLE
Add a generic GPIO driver for Rockchip SoCs and GPIO speed benchmarks

### PR DESCRIPTION
**Pull request by ZhangGaoxing**
1. Add a generic GPIO driver `RockchipDriver` for Rockchip SoCs and some special GPIO drivers like `OrangePi4Driver`, `NanoPiR2sDriver`
2. Add GPIO speed benchmarks

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/iot/pull/1447)

### CODE REVIEW

File: src/devices/Gpio/Drivers/PinState.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Device.Gpio;
+
+namespace Iot.Device.Gpio.Drivers
+{
+    /// <summary>
+    /// Used to describe the pin status inside the driver.
+    /// </summary>
+    public class PinState
```
Comment (krwq) [inline src/devices/Gpio/Drivers/PinState.cs:11]: the description sounds like this should be internal class yet it's public

File: src/devices/Gpio/README.md [hunk 2]
```diff
@@ -1,80 +1,34 @@
...
+The test uses different GPIO drivers to quickly switch the state of GPIO, and uses an oscilloscope to measure the average frequency of GPIO externally.
 
-### Generic GPIO driver: `SunxiDriver`
+### SunxiDriver
 
-```C#
-// Beacuse this is a generic driver, the pin scheme can only be Logical.
-// The base addresses can be found in the corresponding SoC datasheet.
-using GpioController gpio = new GpioController(PinNumberingScheme.Logical, new SunxiDriver(cpuxPortBaseAddress: 0x01C20800, cpusPortBaseAddress: 0x01F02C00));
+Benchmarking with **Orange Pi Zero**, select GPIO 6 (Logical). The operating system is Armbian buster, Linux kernel version is 5.10.16, and .NET version is 5.0.3.
```
Comment (krwq) [inline src/devices/Gpio/README.md:16]: You should add following info here in order for this information to be more useful:
- version/commit hash/date when this was done
- how to re-create the benchmarks (at least high level overview)

File: src/devices/Gpio/README.md [hunk 3]
```diff
@@ -1,80 +1,34 @@
-# Sunxi GPIO Driver for .NET
+# GPIO Driver for Special Chip
```
Comment (krwq) [inline src/devices/Gpio/README.md:1]: I think this label will look really weird on the device listing. I'd recommend to change to something like:

```suggestion
# GpioDriver for other boards
```

File: src/devices/Gpio/Drivers/PinState.cs (hunk 1 above)
Comment (ZhangGaoxing) [reply src/devices/Gpio/Drivers/PinState.cs:11]: I created a field in `RockchipDriver`. I need to access this field in the subclass, so I used protected. If `PinState` is set to internal, there will be an access error. In fact, this class first appeared here, but there is no inheritance involved here https://github.com/dotnet/iot/blob/0a739b805b80b32e71cfb8875657d243b42174ba/src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3LinuxDriver.cs#L709
```C#
public unsafe class RockchipDriver : SysFsDriver
{
    protected IDictionary<int, PinState> _pinModes = new Dictionary<int, PinState>();
}
```

File: src/devices/Gpio/README.md (hunk 2 above)
Comment (ZhangGaoxing) [reply src/devices/Gpio/README.md:16]: I will add the test program to the sample folder

**Comment by pgrawehr** [comment]
@ZhangGaoxing The build is failing due to the StyleCop change in #1444. Please rebase your PR.

**Comment by joperezr** [comment]
hey @ZhangGaoxing looks like something happened when you last rebased your PR and some changes that are not part of your PR. Do you need any help fixing that and only adding your latest commits?

**Comment by ZhangGaoxing** [comment]
Sorry, I seem to have made a mistake, I will submit a new PR🙁

**Pull request closed without merging**